dht/i_partitioner.cc

```diff
@@ -22,6 +22,7 @@
 #include "i_partitioner.hh"
```

Comment:
I think we should move sharding logic out of partitioner completely (far from the `i_partitioner` class and the i_partitioner.hh,cc files).

As you said in the PR cover letter, partitioners should be just `key -> token` functions. Sharding logic says which token belongs to which shard, which is independent of the `key -> token` conversion.

Reply:
I wanted to do that but sharding logic depends on shards_count, sharding_ignore_msb_bits and shard_start. At the moment there's no better place to keep this state than partitioner.